### PR TITLE
[All Hosts] (troubleshooting) resid errors

### PR DESCRIPTION
**Pull request by Rick-Kirkham**
This docs the 32 char limit for resids and fixes [https://github.com/OfficeDev/office-js-docs-pr/issues/2297](https://github.com/OfficeDev/office-js-docs-pr/issues/2297)

### CODE REVIEW

File: docs/reference/manifest/functionfile.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ Specifies the source code file for operations that an add-in exposes in one of t
 * Add-in commands that execute a JavaScript function instead of displaying UI.
 * Keyboard shortcuts that execute a JavaScript function.
 
-The `FunctionFile` element is a child element of [DesktopFormFactor](desktopformfactor.md) or [MobileFormFactor](mobileformfactor.md). The `resid` attribute of the `FunctionFile` element is set to the value of the `id` attribute of a `Url` element in the `Resources` element that contains the URL to an HTML file that contains or loads all  the JavaScript functions used by UI-less add-in command buttons, as defined by the [Control element](control.md).
+The `FunctionFile` element is a child element of [DesktopFormFactor](desktopformfactor.md) or [MobileFormFactor](mobileformfactor.md). The `resid` attribute of the `FunctionFile` element can be no more than 32 characters and is set to the value of the `id` attribute of a `Url` element in the `Resources` element that contains the URL to an HTML file that contains or loads all  the JavaScript functions used by UI-less add-in command buttons, as defined by the [Control element](control.md).
```
Comment (ElizabethSamuel-MSFT) [inline docs/reference/manifest/functionfile.md:15]:
```suggestion
The `FunctionFile` element is a child element of [DesktopFormFactor](desktopformfactor.md) or [MobileFormFactor](mobileformfactor.md). The `resid` attribute of the `FunctionFile` element can be no more than 32 characters and is set to the value of the `id` attribute of a `Url` element in the `Resources` element that contains the URL to an HTML file that contains or loads all the JavaScript functions used by UI-less add-in command buttons, as defined by the [Control element](control.md).
```

File: docs/testing/troubleshoot-development-errors.md [hunk 2]
```diff
@@ -60,6 +60,17 @@ Check the reference documentation for the property to see if it is read only. Al
...
+- The add-in's manifest has a `resid` that is not defined anywhere in the [Resources](../reference/manifest/resources.md) section of the manifest, or there is a mismatch in the spelling of the `resid` between where it is used and where it is defined in the `<Resources>` section.
+- There is a `resid` attribute somewhere in the manifest with more than 32 characters. A `resid` attribute, and the `id` attribute of the corresponding resource in the `<Resources>` section, cannot be more than 32 characters.
+- The add-in has a custom Add-in Command but you are trying to run it on a platform that does not support them. For more information, see [Add-in commands requirement sets](../reference/requirement-sets/add-in-commands-requirement-sets.md).
```
Comment (ElizabethSamuel-MSFT) [inline docs/testing/troubleshoot-development-errors.md:72]:
```suggestion
- The add-in has a custom Add-in Command but you are trying to run it on a platform that doesn't support them. For more information, see [Add-in commands requirement sets](../reference/requirement-sets/add-in-commands-requirement-sets.md).
```

File: docs/reference/manifest/action.md [hunk 3]
```diff
@@ -45,7 +45,7 @@ Required element when **xsi:type** is "ExecuteFunction". Specifies the name of t
 
 ## SourceLocation
 
-Required element when **xsi:type** is "ShowTaskpane". Specifies the source file location for this action. The **resid** attribute must be set to the value of the **id** attribute of a **Url** element in the **Urls** element in the [Resources](resources.md) element.
+Required element when **xsi:type** is "ShowTaskpane". Specifies the source file location for this action. The **resid** attribute can be no more than 32 characters and must be set to the value of the **id** attribute of a **Url** element in the **Urls** element in the [Resources](resources.md) element.
```
[the remark this answers is not in the document]
Comment (Rick-Kirkham) [reply docs/reference/manifest/action.md:48]: I think my wording is clearer. At any rate, there isn't enough of a difference to bother making.

File: docs/reference/manifest/getstarted.md [hunk 4]
```diff
@@ -19,15 +19,15 @@ Provides information used by the callout that appears when the add-in is install
 
 ### Title 
 
-Required. The title used for the top of the callout. The **resid** attribute references a valid ID in the **ShortStrings** element in the [Resources](resources.md) section.
+Required. The title used for the top of the callout. The **resid** attribute references a valid ID in the **ShortStrings** element in the [Resources](resources.md) section and can be no more than 32 characters.
```
[the remark this answers is not in the document]
Comment (Rick-Kirkham) [reply docs/reference/manifest/getstarted.md:22]: I prefer my wording.

File: docs/reference/manifest/action.md [hunk 5]
```diff
@@ -45,7 +45,7 @@ Required element when **xsi:type** is "ExecuteFunction". Specifies the name of t
```
[the remark this answers is not in the document]
Comment (ElizabethSamuel-MSFT) [reply docs/reference/manifest/action.md:45]: @Rick-Kirkham Just confirming that you're opting not to update the ms.date metadata on the changed files?

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Pull request merged**